### PR TITLE
Add gift card update view #2

### PR DESCRIPTION
**Pull request by mmarkusik**
I want to merge this change because it adds info card to gift card details page, as well as some fixes and refactor

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3912 Add Gift Card update view header, basic details & metadata fields](https://app.clickup.com/t/2549495/SALEOR-3912)

File: src/misc.ts [hunk 1]
```diff
@@ -458,3 +458,13 @@ export const transformAddressToAddressInput = (data?: AddressType) => ({
...
 });
+
+export function getFullName<T extends { firstName: string; lastName: string }>(
+  data: T
+) {
+  if (!data) {
+    return "";
+  }
+
+  return `${data.firstName} ${data.lastName}`;
```
Comment (jwm0) [inline src/misc.ts:469]: You're not checking if `firstName` and `lastName` are undefined. The result of this function could be "undefined undefined"

File: src/giftCards/GiftCardUpdatePage/GiftCardUpdateDetailsCard/GiftCardUpdateInfoCard.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        )}
+        <CardSpacer />
+
+        <Label text={intl.formatMessage(messages.usedByLabel)} />
+        {usedBy ? (
+          <Link onClick={() => navigate(customerUrl(usedBy.id))}>
+            {getFullName(usedBy)}
+          </Link>
+        ) : (
+          <Typography>{usedByEmail || PLACEHOLDER}</Typography>
```
Comment (jwm0) [inline src/giftCards/GiftCardUpdatePage/GiftCardUpdateDetailsCard/GiftCardUpdateInfoCard.tsx:136]: Extend `getStringOrPlaceholder` function to take a second `placeholder` argument and use it here

File: src/giftCards/GiftCardUpdatePage/GiftCardUpdateDetailsCard/TimePeriodField.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { useIntl } from "react-intl";
+
+import { timePeriodTextWithSelectFieldMessages as messages } from "./messages";
+
+interface TimePeriodFieldProps<T> {
+  change: FormChange;
+  periodAmount: number;
+  periodType: TimePeriodTypeEnum;
+  amountFieldName: keyof T;
+  typeFieldName: keyof T;
```
Comment (jwm0) [inline src/giftCards/GiftCardUpdatePage/GiftCardUpdateDetailsCard/TimePeriodField.tsx:14]: What's the use of generic typ here? What is T, you never define the original object. `string` seems more fitting

**Review by jwm0 — CHANGES_REQUESTED** [review]

**Review by orzechdev — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**